phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1124,9 +1153,19 @@ private void waitForPreviousConcurrentBatch(TableName table, BatchMutateContext
         boolean done = true;
         for (BatchMutateContext lastContext : context.lastConcurrentBatchContext.values()) {
             BatchMutatePhase phase = lastContext.getCurrentPhase();
-
-            if (phase == BatchMutatePhase.PRE) {
+            if (phase == BatchMutatePhase.FAILED) {
+                done = false;
+                break;
+            } else if (phase == BatchMutatePhase.PRE) {
                 CountDownLatch countDownLatch = lastContext.getCountDownLatch();
+                if (countDownLatch == null) {
+                    // phase changed from PRE to either FAILED or POST
+                    if (phase == BatchMutatePhase.FAILED) {
```

Comment:
Since `phase` is not getting updated, it will always be `PRE` here. Maybe we need to change this condition to `if (lastContext.getCurrentPhase() == BatchMutatePhase.FAILED)`?